background/pool.ts

```diff
@@ -21,35 +26,56 @@ export class RequestStoragePool implements RequestPool {
     accountId: string
     networkId: string
   }) {
-    const { request, accountId, networkId } = data
-
-    if (request instanceof Eip712Request) {
-      throw new Error("EIP-712 is not yet supported")
-    }
-
-    const id = uuidv4()
+    const request = this.transformPendingRequest(data)
 
     this.storage.set((state) => {
-      state.pendingRequests.push({
-        type: RequestType.Transaction,
-        id,
-        createdAt: Date.now(),
-        accountId,
-        networkId,
-        ...request.unwrap()
-      })
+      state.pendingRequest[request.id] = request
     })
 
-    return id
+    return request.id
+  }
+
+  public wait(requestId: string) {
```

Comment:
Comparing with [`RequestHandler`](https://github.com/pilagod/waallet-extension/blob/5688d0b4db5e1999d5b9c8f3cae9dc2dfd313c6d/packages/waallet/background/pool/request/handler.ts#L17) and `RequestStoragePool`, I found the responsibility of `send` and `wait` are different.
- in `RequestStoragePool`, `send` will not process the request, but the one in `RequestHandler` will handle the request and store the promise.

Reply:
Good catch, you're right. The `RequestHandler` is actually written for test. In tests, we don't have storage, and we usually want to get result immediately, so `RequestHandler` will handle the request right away and send back result to tests as soon as possible.

In terms of the project structure, `RequestHandler` would be more suitable to put into somewhere like `packages/util/testing` to reveal more of its intension.